### PR TITLE
Fix ReDos attack in #117

### LINKED ISSUE #117: SECURITY: bad regex pattern in 'sklearn/externals/_arff.py' will cause 'ReDos' security problem.
Posted at https://github.com/scikit-learn/scikit-learn/issues/19522 by @leveryd, `_RE_TYPE_NOMINAL` presents a ReDos risk.

### PR DESCRIPTION
**Pull request by jnothman**
Fixes #117. Without this patch, the test hangs.

### CODE REVIEW

**Comment by jnothman** [comment]
And the solution is more efficient than the original due to regex overkill.

File: tests/test_decode_attribute_types.py [hunk 1]
```diff
@@ -123,6 +123,10 @@ def test_nominal(self):
         self.assertEqual(result[1][2], expected[1][2])
         self.assertEqual(result[1][3], expected[1][3])
 
+        # Check non-regression of ReDos raised in https://github.com/renatopp/liac-arff/issues/117
```
Comment (mfeurer) [inline tests/test_decode_attribute_types.py:126]: Sorry, I don't see how this is checked at the moment, could you please explain this?

**Review by mfeurer — COMMENTED** [review]
Thanks a lot for forwarding this issue and providing a fix. I have a hopefully simple question before we can merge this.

@ogrisel I don't understand your comment in #117, could you please explain?

File: tests/test_decode_attribute_types.py (hunk 1 above)
Comment (jnothman) [reply tests/test_decode_attribute_types.py]: Sorry, a couple of lines must have gone missing in commit.

**Review by jnothman — COMMENTED** [review]
Clearer now

**Comment by jnothman** [comment]
sorry, those commits were a bit confused, trying to remember what I'd tested before

**Comment by mfeurer** [comment]
Alright, thanks a lot for fixing the issue and updating the unit test.

**Pull request merged**